### PR TITLE
misc: remove production/ in favor of operations/

### LINKED ISSUE #6077: Remove the `production/` folder in favor of `operations/`
### Background

The `production/` folder is composed of four pieces:

* `grafana-agent-mixin/`: A Jsonnet mixin for static mode monitoring
* `kubernetes/`: Kubernetes manifests to deploy the agent in static mode
* `operator/`: CRDs and manifests for deploying the operator 
* `tanka/`: Tanka configs for deploying both static mode and the operator 

The `production/` folder has been effectively abandoned over time in favor of the new `operations/` folder, which contains:

* `agent-flow-mixin/`: A Jsonnet mixin for flow mode monitoring.
* `helm/`: Helm chart(s) for Grafana Agent.
* `river-jsonnet/`: A Jsonnet library for writing River in Jsonnet. 

Having both of these directories is confusing, as it's not obvious (unless you look at Git history) which is newer. Additonally, some of the subdirectories in `production/` are effectively abandoned. 

`production/` should be merged into `operations/`, and abandoned subdirectories should be removed. 

### Proposal

Merge non-abandoned subdirectories from `production/` into `operations/`: 

- [ ] Move `production/grafana-agent-mixin` to `operations/agent-static-mixin` 
- [ ] Remove `production/kubernetes` (deprecated by `operations/helm`)
- [ ] Move `production/operator` to `operations/agent-static-operator`, and update the Makefile at the root of the repository to generate CRD manifests in the new directory.
- [ ] Remove `production/tanka` (deprecated by `operations/helm`) 

Care must be taken before removing these directories to ensure that no documentation points at this directory either from this repo or within Grafana Cloud instructions.

### PR DESCRIPTION
**Pull request by rfratto**
This commit entirely removes the production/ folder in favor of operations/; all known valid external references to the old production folder have been removed.

Closes #6077.

### CODE REVIEW

File: .drone/pipelines/test.jsonnet [hunk 1]
```diff
@@ -50,23 +50,6 @@ local pipelines = import '../util/pipelines.jsonnet';
...
-      image: build_image.linux,
-
-      commands: [
-        'make generate-manifests',
-        'ERR_MSG="The environment manifests are out of date. Please run \'make generate-manifests\' and commit changes!"',
-        // "git status --porcelain" reports if there's any new, modified, or deleted files.
-        'if [ ! -z "$(git status --porcelain)" ]; then echo $ERR_MSG >&2; exit 1; fi',
-      ],
-    }],
-  },
```
Comment (rfratto) [inline .drone/pipelines/test.jsonnet:68]: We have no static manifests anymore, so this job has been removed.

File: Makefile [hunk 2]
```diff
@@ -314,13 +313,6 @@ else
 	bash ./operations/helm/scripts/rebuild-tests.sh
 endif
 
-generate-manifests:
-ifeq ($(USE_CONTAINER),1)
-	$(RERUN_IN_CONTAINER)
-else
-	cd production/kubernetes/build && bash build.sh
-endif
```
Comment (rfratto) [inline Makefile:322]: We have no static manifests to generate anymore, so this target has been removed.

**Review by tpaschalis — APPROVED** [review]
Rubber stamping but I can also take a more detailed look tomorrow if you think it's worthwhile.

**Comment by rfratto** [comment]
> Rubber stamping but I can also take a more detailed look tomorrow if you think it's worthwhile.

Nah, that's OK. I'm pretty sure I got all of the references and if I did miss something, it's probably minor and we can fix things as they come up.

**Pull request merged**